staging/src/k8s.io/apimachinery/third_party/forked/golang/json/fields.go

```diff
@@ -52,9 +52,41 @@ func LookupPatchMetadata(t reflect.Type, jsonField string) (reflect.Type, string
 		}
 		patchStrategy := tjf.Tag.Get("patchStrategy")
 		patchMergeKey := tjf.Tag.Get("patchMergeKey")
-		return tjf.Type, patchStrategy, patchMergeKey, nil
+		hasReplacekeys, otherPatchStrategy, err := extractReplaceKeysPatchStrategy(patchStrategy)
+		if err != nil {
+			return nil, false, "", "", err
+		}
+		return tjf.Type, hasReplacekeys, otherPatchStrategy, patchMergeKey, nil
+	}
+	return nil, false, "", "", fmt.Errorf("unable to find api field in struct %s for the json field %q", t.Name(), jsonField)
+}
+
+// extractReplaceKeysPatchStrategy process patch strategy, which is a string may contains multiple
+// patch strategies seperated by "|". It returns a boolean var indicating if it has replaceKeys strategies
+// and a string or other strategy.
+func extractReplaceKeysPatchStrategy(patchStrategy string) (bool, string, error) {
+	replaceKeysPatchStrategy := "replaceKeys"
```

Comment:
I hesitated to use `strategicpatch.ReplaceKeysDirective`, because I'm worrying about introducing dependency of `k8s.io/apimachinery/third_party/forked/golang/json/fields.go` to `k8s.io/apimachinery/util/strategicpatch`. But it seems to be OK.